### PR TITLE
ShowCaseView on Interactive Book

### LINKED ISSUE #105: [Interactive Book] Showcase and brief buttons and toggles at first launch.
**Is your feature request related to a problem? Please describe.**
Someone new to Interactive Book will find all the options like Table of Contents, Chapters and Sections confusing, we should provide an introductory briefing about Interactive Book and it's options.

**Describe the solution you'd like**
Showcasing the App at first launch using Flutter package would be ideal, this will involve designing the Showcase wrapper widget on top of our built-in widgets.

**Additional context**
image

https://pub.dev/packages/showcaseview

### PR DESCRIPTION
**Pull request by aman-singh7**
Fixes #105 

Describe the changes you have made in this PR -
I have added the `ShowCaseView` package to showcase the button and toggles on the Interactive book and model for storing the state of the button pressed. 

Screenshots of the changes (If any) -

https://user-images.githubusercontent.com/77198905/142761255-e301a710-a738-4a09-91ce-238a765f150e.mp4

Note: Please check Allow edits from maintainers. if you would like us to assist in the PR.

### CODE REVIEW

**Comment by aman-singh7** [comment]
@manjotsidhu Implementation of showcase widget is done from side. I am currently working on the issues of the widget test. Please lemme know if there are any changes required.

**Comment by manjotsidhu** [comment]
Thank you for your contributions @aman-singh7, Initial impressions of the code looks good. I will review and test it very soon.

**Comment by aman-singh7** [comment]
@manjotsidhu widget test -[x].

**Comment by aman-singh7** [comment]
@manjotsidhu could you give some info to fix the CI failing?  :sweat_smile:

**Comment by manjotsidhu** [comment]
> @manjotsidhu could you give some info to fix the CI failing? 😅

The CI is not failing due to your changes.

**Comment by aman-singh7** [comment]
@manjotsidhu can you now run the workflow? I have rebased the branch with master initially I just merged the master.

File: lib/models/ib/ib_showcase.dart [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import 'package:flutter/material.dart';
+
+class IBShowCase {
+  bool nextButton, prevButton, contentButton, drawerButton;
+
+  IBShowCase({
+    @required this.nextButton,
+    @required this.prevButton,
+    @required this.contentButton,
```
Comment (manjotsidhu) [inline lib/models/ib/ib_showcase.dart:11]: since this variable is for table of contents, we can enhance it and name it as `tocButton`

**Comment by manjotsidhu** [comment]
While testing, after showcasing next and previous buttons, it didn't show drawer showcase until I went to next/previous page. Is this intended ?

File: lib/models/ib/ib_showcase.dart (hunk 1 above)
Comment (aman-singh7) [reply lib/models/ib/ib_showcase.dart:11]: okk. :+1:

**Comment by aman-singh7** [comment]
> While testing, after showcasing next and previous buttons, it didn't show drawer showcase until I went to next/previous page. Is this intended ?

The logic to trigger the Drawer Showcase is currently written inside `ValueListenableBuilder`. So, it will triggered (if not pressed earlier) when there is a content button on that screen

**Comment by manjotsidhu** [comment]
> > While testing, after showcasing next and previous buttons, it didn't show drawer showcase until I went to next/previous page. Is this intended ?
> 
> The logic to trigger the Drawer Showcase is currently written inside `ValueListenableBuilder`. So, it will triggered (if not pressed earlier) when there is a content button on that screen

I think I opened drawer button earlier so that might be the reason.

**Comment by manjotsidhu** [comment]
I tested again, it triggered drawer showcase an reaching next pages.

**Comment by manjotsidhu** [comment]
https://user-images.githubusercontent.com/22657113/143593331-1a636fc5-a79f-4b81-a4f6-528f210d026f.mp4

**Comment by aman-singh7** [comment]
> I tested again, it triggered drawer showcase an reaching next pages.

I am updating the state of button clicked in `onTargetClick`. Until and unless button is not pressed in showcaseview it is going to show.

**Comment by aman-singh7** [comment]
@manjotsidhu In what sequence do you want the buttons to be showcased?

**Comment by aman-singh7** [comment]
> > While testing, after showcasing next and previous buttons, it didn't show drawer showcase until I went to next/previous page. Is this intended ?
> 
> The logic to trigger the Drawer Showcase is currently written inside `ValueListenableBuilder`. So, it will triggered (if not pressed earlier) when there is a content button on that screen

Update:- If not pressed implies if not pressed when showcased.

**Comment by manjotsidhu** [comment]
> > > While testing, after showcasing next and previous buttons, it didn't show drawer showcase until I went to next/previous page. Is this intended ?
> > 
> > 
> > The logic to trigger the Drawer Showcase is currently written inside `ValueListenableBuilder`. So, it will triggered (if not pressed earlier) when there is a content button on that screen
> 
> *If not pressed implies if not pressed when showcased.

Then the drawer showcase should be independent of toc being there or not. Drawer showcase is important initially.

**Comment by aman-singh7** [comment]
> > > > While testing, after showcasing next and previous buttons, it didn't show drawer showcase until I went to next/previous page. Is this intended ?
> > > 
> > > 
> > > The logic to trigger the Drawer Showcase is currently written inside `ValueListenableBuilder`. So, it will triggered (if not pressed earlier) when there is a content button on that screen
> > 
> > 
> > *If not pressed implies if not pressed when showcased.
> 
> Then the drawer showcase should be independent of toc being there or not. Drawer showcase is important initially.

Cool.. I will seperate it from toc.

**Comment by aman-singh7** [comment]
https://user-images.githubusercontent.com/77198905/144548999-74d18663-439d-4344-b189-d7785c490734.mp4

@manjotsidhu I have updated the logic of showcasing. I have a doubt that the current implementation forces the user to click on the button when showcased otherwise it is going to showcase after every page changed. Do we need to force the user or not showcase the button which are previously clicked.

**Comment by manjotsidhu** [comment]
> 
> https://user-images.githubusercontent.com/77198905/144548999-74d18663-439d-4344-b189-d7785c490734.mp4
> 
> @manjotsidhu I have updated the logic of showcasing. I have a doubt that the current implementation forces the user to click on the button when showcased otherwise it is going to showcase after every page changed. Do we need to force the user or not showcase the button which are previously clicked.

We should not ideally force the user to click the button, if the user dismisses the showcase then it is safe to say that our showcasing is done and no need to show showcase again.

**Comment by aman-singh7** [comment]
@manjotsidhu can you now test the showcase?

**Comment by Nitish145** [comment]
@manjotsidhu is the showcase working fine?

**Comment by manjotsidhu** [comment]
Apologies for late reply, thanks for your contribution. I will test the showcase tonight but we will need to fix for iOS as well in order to merge. Would have been better if you have made a new PR for 2.8 since this PR's scope was showcase and we are also including fixes for 2.8.

**Comment by aman-singh7** [comment]
> Apologies for late reply, thanks for your contribution. I will test the showcase tonight but we will need to fix for iOS as well in order to merge. Would have been better if you have made a new PR for 2.8 since this PR's scope was showcase and we are also including fixes for 2.8.

Ok, I will make a separate PR for 2.8

**Review by manjotsidhu — APPROVED** [review]
✅ Tested and working perfectly on Android
✅ Implementation LGTM 👍

**Pull request merged**